lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -4,50 +4,31 @@ class Pod < KubernetesResource
     TIMEOUT = 10.minutes
     SUSPICIOUS_CONTAINER_STATES = %w(ImagePullBackOff RunContainerError ErrImagePull).freeze
 
-    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
-      @name = name
-      @namespace = namespace
-      @context = context
-      @file = file
+    def initialize(namespace:, context:, definition:, logger:, parent: nil, deploy_started: nil)
       @parent = parent
-      @logger = logger
+      @deploy_started = deploy_started
+      @containers = definition["spec"]["containers"].map { |c| c["name"] }
```

Comment:
Hm, maybe I should raise here like I'm now doing in the base class if metadata.name is missing. This is similarly a user error that should be exposed as a template problem, not a stacktrace.